pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -251,19 +252,33 @@ func RunStartBuild(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra
 	if follow {
 		wg.Add(1)
 		go func() {
-			defer wg.Done()
+			// if --wait option is set, then don't wait for logs to finish streaming
+			// but wait for the build to reach its final state
+			if waitForComplete {
+				wg.Done()
+			} else {
+				defer wg.Done()
+			}
 			opts := buildapi.BuildLogOptions{
 				Follow: true,
 				NoWait: false,
 			}
-			rd, err := client.BuildLogs(namespace).Get(newBuild.Name, opts).Stream()
-			if err != nil {
-				fmt.Fprintf(cmd.Out(), "error getting logs: %v\n", err)
-				return
-			}
-			defer rd.Close()
-			if _, err = io.Copy(out, rd); err != nil {
-				fmt.Fprintf(cmd.Out(), "error streaming logs: %v\n", err)
+			for {
```

Comment:
I think we need a cap on this retry.  otherwise if something goes wrong creating the build, we could sit her forever with no message to the user about what is happening.